src/native/minipal/getexepath.h

```diff
@@ -89,6 +89,8 @@ static inline char* minipal_getexepath(void)
     }
 
 #if HAVE_GETAUXVAL && defined(AT_EXECFN)
+    errno = 0;
+
     // fallback to AT_EXECFN, which does not work properly in rare cases
     // when .NET process is set as interpreter (shebang).
     const char* exePath = (const char *)(getauxval(AT_EXECFN));
```

Comment:
Should we check the `errno` after the call to `getauxval` at all? The doc says that `getauxval` either returns a value or 0 when the value doesn't exist.

Reply:
Yes, makes sense. This is the relevant text from manpage:

```
RETURN VALUE

       On success, getauxval() returns the value corresponding to type.
       If type is not found, 0 is returned.

ERRORS

       ENOENT (since glibc 2.19)
              No entry corresponding to type could be found in the
              auxiliary vector.

...

BUGS

       Before the addition of the ENOENT error in glibc 2.19, there was
       no way to unambiguously distinguish the case where type could not
       be found from the case where the value corresponding to type was
       zero.
```

In our case, we want to return NULL in all error cases.